### PR TITLE
mmc: Fix for bkops-status read fail on Castor

### PR DESCRIPTION
**Pull request by rcstar6696**
This fixes mmc1: mmc_start_bkops: Failed to read bkops status: 1

### CODE REVIEW

**Comment by rcstar6696** [comment]
ping @kholk

File: drivers/mmc/core/core.c [hunk 1]
```diff
@@ -1519,15 +1519,20 @@ int mmc_read_bkops_status(struct mmc_card *card)
...
+#ifdef CONFIG_MACH_SONY_CASTOR
+	err = mmc_send_ext_csd(card, ext_csd);
+#else	
 	if (mmc_card_cmdq(card))
 		err = mmc_cmdq_halt_on_empty_queue(card->host);
 
 	if (!err) {
 		err = mmc_send_ext_csd(card, ext_csd);
-
+
```
Comment (erikcas) [inline drivers/mmc/core/core.c]: oops

**Comment by kholk** [comment]
Looks fine, but should be tested on all devices, and eventually done with DT compatible property only for devices who need it, since the ones that don't need it should be the majority.

**Comment by rcstar6696** [comment]
it seems that only castor needs it noone saw the issue before on 1.3.3
i tested it on sirius too :)

**Comment by kholk** [comment]
Rhine has the same issue

**Comment by alviteri** [comment]
maybe is my fault... but I have grab 3 dmesg in some situations but I can't see this error in honami :|

**Comment by kholk** [comment]
The code is new, only for new eMMCs and should be enabled only for new devices which support the new eMMC standard, which are Dora and Suzu. In this kernel we support only Suzu, so perhaps you should ifdef for suzu and leave the new code disabled for all other devices. It's safe.

**Comment by jerpelea** [comment]
plase add a barier for all 8974 devices (both rhine and shinano)

**Comment by rancidfrog** [comment]
Aries,  8974 has following issue, related?

```
[16384.419460] mmc1: mmc_set_auto_bkops: error in setting auto bkops to 1 (-74)
[16384.426068] mmc1: mmc_init_card: Failed to enable auto-bkops: err: -74
[16384.520883] android_usb gadget: high-speed config #1: fe8050c8.android_usb
[16384.521012] dwc3 f9200000.dwc3: Avail curr from USB = 500
[16384.521500] android_work: android_work: sent uevent USB_STATE=CONNECTED
```

**Comment by alviteri** [comment]
https://github.com/sonyxperiadev/kernel/pull/808

**Pull request closed without merging**